### ISSUE TITLE
Does LRSequential need to update in train_imagenet.py?

### ISSUE THREAD

**Issue by PistonY** [dmlc/gluon-cv#699]
https://github.com/dmlc/gluon-cv/blob/9d67b4343eb9c8ed4b979bdc9d74124bc50196ef/scripts/classification/imagenet/train_imagenet.py#L407

Or this line would lead to a bug
```python
def update(self, num_update):
```

**Comment by ljk628** [comment]
Having the same problem after updating gluon-cv. I think it should be changed to 
`lr_scheduler.update(num_batches*epoch + i)`.

**Comment by PistonY** [comment]
Just del the line should be OK.
It needs not to update manually.

**Comment by hetong007** [comment]
Thank you for the catch! It might due to a careless conflict merge from me.
Should be fixed by https://github.com/dmlc/gluon-cv/pull/712

**Comment by PistonY** [comment]
@hetong007 Hi, if you just delete the line, I don't know whether '--resume-XX' can work totally right, LRSequential may return a wrong learning rate?

**Comment by hetong007** [comment]
@PistonY If I understand your question correctly, you don't need to worry. `Trainer` takes care of the learning rate update by implicitly calling `update` in each `trainer.step()`.

**Comment by PistonY** [comment]
@hetong007 Oh, you mean that when saving the Trainer status by calling `trainer.save_states,` the number of `calling update in each trainer.step()` will be saved too? And when I read it back, `Trainer` would know the learning rate I saved before?

**Comment by hetong007** [comment]
@PistonY Every `trainer.step()` triggers the `lr_scheduler.update()` implicitly. So whether we call `save_states()` doesn't matter, as `trainer.learning_rate` has been changed already during the training process.

**Comment by PistonY** [comment]
I knew that.I mean your train_imageney.py have a feature of ‘resume ’ which make me could save status out in half way of training and then load it back continues training.And I want to know when I load status back the lr is correct or not(it should be the same as I save the status before, especially when I use cosine scheduler)

**Comment by hetong007** [comment]
@PistonY Sorry for the late reply. Seems you don't need to worry about that:

1. The current lr scheduler calculates the lr value depending on `num_update`: https://github.com/dmlc/gluon-cv/blob/master/gluoncv/utils/lr_scheduler.py#L122
2. `num_update` is a property from optimizers and will be saved with `save_states`: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/optimizer/optimizer.py#L193

**Comment by PistonY** [comment]
@hetong007 Thanks for you patient reply, it's very careful feature.
